mlrun/runtimes/nuclio/application/application.py

```diff
@@ -506,9 +529,11 @@ def _configure_application_sidecar(self):
 
         if self.status.container_image:
             self.from_image(self.status.container_image)
-            # nuclio implementation detail - when providing the image and emptying out the source code,
+            # nuclio implementation detail - when providing the image and emptying out the source code and build source,
             # nuclio skips rebuilding the image and simply takes the prebuilt image
             self.spec.build.functionSourceCode = ""
+            self.status.application_source = self.spec.build.source
+            self.spec.build.source = ""
```

Comment:
How about overriding the `from_image` method for application runtime and putting everything there?
Then we can call it independently if we want and it will have the same effect